closed/make/Main.gmk

```diff
@@ -63,7 +63,7 @@ test-image : test-image-openj9
 test-image-openj9 : images
 	@+$(MAKE) $(MAKE_ARGS) -f $(SRC_ROOT)/closed/TestImage.gmk
 ifneq ($(COMPILE_TYPE), cross)
-	$(JRE_IMAGE_DIR)/bin/java -version 2>&1 | $(TEE) $(IMAGES_OUTPUTDIR)/test/openj9/java-version.txt
+	rc=0; $(JRE_IMAGE_DIR)/bin/java -version >$(IMAGES_OUTPUTDIR)/test/openj9/java-version.txt 2>&1 || rc=$$?; $(CAT) $(IMAGES_OUTPUTDIR)/test/openj9/java-version.txt; exit "$$rc"
```

Comment:
Sorry to pick nits, but why the quotes in `exit "$$rc"` (if they're not needed in `rc=$$?`, they're not needed here either)?

Reply:
That's a habit of mine. I think the way I'd put it is that if you see `$?`, quotes are never necessary, and you don't need to read any other part of the script to figure that out. For `$rc`, we can reason that they quotes are unnecessary, but we have to do that reasoning, and with later changes that reasoning could conceivably get out of date (though it's admittedly unlikely). So I always quote my own variables.

If for some reason these two variable uses absolutely must be consistent (i.e. both quoted or both unquoted), then I'll add quotes around `$?`. But if possible I'd prefer not to have to make that change. I don't think it's very helpful, and it's extremely tedious to make another trivial change in these 6 PRs and then check all of them to make sure I'm still getting the behaviour I expect.

Reply:
`$?` should expand to a number and `exit` expects a single, numeric argument, so there's not needed in either place: My preference is that we don't use quotes here. If you want to use quotes, please do so consistently (i.e. both places).

Reply:
Fine

Reply:
Perhaps I should have insisted on removing the quotes; they're only used in places where they're provably unnecessary (and it's still inconsistent: the other assignment to `rc` doesn't have them). I don't think it's worth another round of changes, so let's declare victory.

Reply:
Expanding variables outside of double-quoted strings as much as possible is an error-prone way to write shell scripts. The only reliable way to consistently treat uses of variables (w.r.t. quoting) is to always quote them. That's why that's my general rule (though I sometimes make an exception for `$?` only because it's obvious that it's OK even without reading any other part of the script).

And I disagree that my updated code is still inconsistent in this respect. The literal string `0` is unquoted, whereas uses of variables are quoted. It's irrelevant that both a literal and a variable occur in assignments.

I can at least agree with this though:
> I don't think it's worth another round of changes, so let's declare victory.